lib/matplotlib/axes/_base.py

```diff
@@ -2116,7 +2117,9 @@ def _update_patch_limits(self, patch):
         if (isinstance(patch, mpatches.Rectangle) and
                 ((not patch.get_width()) and (not patch.get_height()))):
             return
-        vertices = patch.get_path().vertices
+        vertices = np.array([v for s in patch.get_path().iter_segments()
+                             if s[1] != mpath.Path.CLOSEPOLY
+                             for v in s[0]]).reshape([-1, 2])
```

Comment:
Both `.vertices` and `.codes` should be arrays (or `None`), so it should be possible to do this much faster using array indexing instead of a comprehension.

Reply:
Probably I'm displaying my ignorance here (I'm unclear on what exactly you're suggesting), but the challenges are that you do not know how many codes are not CLOSEPOLY, and that there's not a 1-to-1 correspondence between .verticies and .codes.
On another note, I'm blindly copying the style of what is there (using the vertices array); what is unclear to me is why the original code did not just use patch.get_path().get_extents(transform=???).

Reply:
For arrays, you can do `x[y != z]` instead of `[i for i, j in zip(x, y) if j != z]`.

Reply:
That answers my concern about the unknown number that are CLOSEPOLY, but the problem is that the .verticies array and .codes arrays are different sizes because CURVE3 and CURVE4 codes have 2 and 3 vertices in the vertices array. iter_segments() takes care of the correspondence, returning the appropriate number of coordinates for each element of codes.
And reading the documentation again it looks like I also need to exclude STOP codes...

Reply:
The `Path` docs say that `vertices` and `codes` must be the same length.

Reply:
My bad, I didn't read the docs carefully enough. I've pushed something that should be better.